src/commands/deploy.ts

```diff
@@ -66,6 +66,22 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         }
     }
 
+    let siteConfig: WebSiteModels.SiteConfigResource | undefined;
+
+    if (!fsPath) {
+        // we can only get the siteConfig if the entry point was a treeItem
+        siteConfig = node ? await node.root.client.getSiteConfig() : undefined;
+
+        if (siteConfig && javaUtils.isJavaRuntime(siteConfig.linuxFxVersion)) {
+            const fileExtension: string = javaUtils.getArtifactTypeByJavaRuntime(siteConfig.linuxFxVersion);
+            fsPath = await javaUtils.showQuickPickByFileExtension(context, `Select the ${fileExtension} file to deploy...`, fileExtension);
+            // if there is a siteConfig, then there is a node
+            await javaUtils.configureJavaSEAppSettings(nonNullValue(node));
+        } else {
+            fsPath = await workspaceUtil.showWorkspaceFoldersQuickPick("Select the folder to deploy", context, constants.configurationSettings.deploySubpath);
```

Comment:
This won't work for Java projects if they use the command palette or blue up arrow. One idea - you could combine `showQuickPickByFileExtension` and `showWorkspaceFoldersQuickPick` to work in both cases. In other words, show both the workspace folders _and_ any *.war/*.jar files